### PR TITLE
Update Islandora Core for phpcs sniffs

### PR DESCRIPTION
**Pull request by jonathangreen**
🚨 **This needs some feedback and collaboration before being merged** 🚨 

## JIRA Ticket

https://jira.duraspace.org/browse/ISLANDORA-2106

## What does this Pull Request do?

Decided to take a crack at this since I had a few spare hours on a slow Thursday evening.

This brings the coding standards in core Islandora into line with the new sniffs from `coder` and `phpcs`, this is part of the work discussed in the committers meeting the other day.

This is meant to be an example for what is to be done in other modules and to discuss how we are going to implement this.

## What's new?

- Add a new script to the test directory to install phpcs and the sniffs. `phpcs_setup.sh`
  - This is intended to be used in all the child modules.
- Update the coding standards in the module to conform to the new phpcs sniffs.

## Additional Notes:

Some things for discussion: 
- The new sniffs include using type hints for types defined in the phpdoc comments. I added all of these type hints, except where we obviously lied in the phpdoc, then I changed the phpdoc instead. These new type hints have the ability to blow up existing code if they are wrong. Are we okay with that? 
- What sniffs and configuration do we want to use? 
  - I used both `drupal` and `drupalpractice` here, and I think that is good, but we should probably discuss.
  - I excluded `info`, `md` and `txt` files. 
    - `md` and `txt` because the new sniffs mandate no lines are longer then 80 characters in these files, and we don't do this. We do the same thing in CLAW.
    - `info` because the sniff enforces no `version` in the `info` file. We don't package on drupal.org, so this doesn't really apply to us.
- I use [ignore comments](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#ignoring-parts-of-a-file) a couple places. I think they should be used sparingly, but will probably have to be used when sniffer wants us to change class names etc.
- This doesn't work right now for PHP 5.3.3. I tried moving the PR to 5.3 to try to get it working, and its still failing, both the `phpcs` part and the tests. I've pretty much given up on fighting with 5.3 every time I try to do something, but do any @Islandora/7-x-1-x-committers want to pick up the torch and try to slay 🔪 👹 the PHP 5.3 demon here?
  - Feel free to push to my branch or make a PR on my PR if you are interested in collaborating.

## Interested parties

@Islandora/7-x-1-x-committers

### CODE REVIEW

**Comment by DiegoPino** [comment]
🚨 🚨 🚨 🚨 🚨  ⚠️   🚨 🚨 🚨 🚨  (love this) will get into your pull after release if that sounds good. Love this (2x)!

**Comment by whikloj** [comment]
@jonathangreen Looks like this did not register the directories. Maybe for PHP 5.3 (which might work for all) you could add
`phpcs --config-set installed_paths ~/.composer/vendor/drupal/coder/coder_sniffer`

File: islandora.module [hunk 1]
```diff
@@ -367,7 +369,7 @@ function islandora_menu() {
     'title' => 'Event Status',
     'page callback' => 'islandora_event_status',
     'type' => MENU_CALLBACK,
-    'access callback' => TRUE,
+    'access arguments' => array(ISLANDORA_VIEW_OBJECTS),
```
Comment (willtp87) [inline islandora.module]: Functional change, probably requires release documentation.

**Comment by willtp87** [comment]
I ran a series of isolation tests and it seems like our tests won't run in the environment on `PHP 5.3.29` (the version for `5.3`) and the CS work won't install on  `5.3` and is not complete on `5.3.29`.  I think the path of least resistance here is to simply not run the new CS on 5.3 and continue to run the tests as is. It retains test coverage for PHP 5.3 and we can rely on the CS from more advanced versions.

**Comment by jonathangreen** [comment]
Lack of interest, so I'm going to close this one. 

Feel free to reopen in the future if someone is looking at coding standards.

**Pull request closed without merging**